teachertool/src/transforms/removeCriteriaFromChecklist.ts

```diff
@@ -14,7 +14,10 @@ export function removeCriteriaFromChecklist(criteriaInstanceId: string) {
 
     const newChecklist = {
         ...teacherTool.checklist,
-        criteria: teacherTool.checklist.criteria.filter(c => c.instanceId !== criteriaInstanceId),
+        criteria: teacherTool.checklist.criteria.filter(c =>
+            c.instanceId !== criteriaInstanceId ||
+            c.instanceId === criteriaInstanceId && !c.deleted
```

Comment:
Hmm, I'm wondering if there's a scenario where we would want this action to work even for non-deleted criteria (in which case we could have a separate action `removeDeletedCriteriaFromChecklist` that checks for deleted before running remove, or just an `onlyIfDeleted` parameter passed into this action). But I can't think of a realistic scenario at the moment so maybe it's fine...